### ISSUE TITLE
Singularity V3.6.0 breaks  mounting the overlay

### ISSUE THREAD

**Issue by maxadam2017** [prioux/withoverlay#1]
Using Singularity V3.6 the script fails with with:
```Could not create overlay work dir. If using an overlay image ensure it contains 'upper' and 'work' directories
FATAL:   container creation failed: hook function for tag prelayer returns error: failed to create
```
 Singularity V2.6.0 with an overlay requires a "work" directory:
`mkdir -p overlay/upper overlay/work`
See https://github.com.cnpmjs.org/hpcng/singularity/issues/5430 for details.

Using Singularity V3.5 the overly mounts, but it's only writable by root.

**Comment by prioux** [comment]
Right. Got a workaround for this. the mkfs.ext3 command on ubuntu has a "-d" option that allows the filesystem to be seeded at creation time. Since singularity wants a /work and /upper directories int he overlay, we'll have to add them.

**Comment by prioux** [comment]
On CentOS, -d doesn't exist... :-(
